packages/compass-import-export/src/import/analyze-csv-fields.spec.ts

```diff
@@ -180,4 +182,40 @@ describe('analyzeCSVFields', function () {
     // signals that it was aborted and the results are therefore incomplete
     expect(result.aborted).to.equal(true);
   });
+
+  it('does not mind windows style line breaks', async function () {
+    const text = await fs.promises.readFile(fixtures.csv.good_commas, 'utf8');
+    const input = Readable.from(text.replace(/\n/g, '\r\n'));
+
+    const result = await analyzeCSVFields({
+      input,
+      delimiter: ',',
+    });
+    expect(Object.keys(result.fields)).to.deep.equal(['_id', 'value']);
+  });
+
+  it('does not mind if a file is not valid utf8', async function () {
+    const latin1Buffer = buffer.transcode(
+      Buffer.from('ê,foo\n1,2'),
+      'utf8',
+      'latin1'
+    );
```

Comment:
This is equivalent to:

```suggestion
    const latin1Buffer = Buffer.from('ê,foo\n1,2', 'latin1');
```